36-vending-machine-monitor/README.md

```diff
@@ -0,0 +1,319 @@
+
+```json
+{ "active": false, "text": "Power restored" }
+```
+
+#### Battery Low Alerts
+
+When running from battery, a battery low alert is sent when the battery voltage falls below
+`BATTERY_LOW_VOLTAGE` volts. (Default setting is 3.2v.)
+
+```json
+{ "active": true, "text": "Battery low" }
+```
+
+When power resumes, the battery low alert is cleared.
+
+```json
+{ "active": true, "text": "Battery charging" }
+```
+
+#### Sensor offline alerts for each column
+
+When the app is unable to read from the distance sensor for a particular column, it sends a Sensor Offline alert.
+
+```json
+{ "active": true, "text": "Distance sensor offline", "col": 1 }
+```
+
+When the app is able to read from the sensor again, it sends clears the alert.
+
+```json
+{ "active": false, "text": "Distance sensor online", "col": 1 }
+```
+
+#### Vending notifications
+
+Vending notifications are sent to `vending.qo`. These notifications, one for each column, include details of the state of the dispensing column, the number if items in the column and information about what has changed since the previous notification.
+
+A typical event sent after an item has been dispensed looks like this:
+
+```json
+{ "col":1, "name":"cola", "state": "good", "dist_mm": 152, "items": 3, "changes": "items,state", "text": "Item dispensed." }
+```
+
+* `col`: The column number.
+* `name`: The name of the column (from the `col_X_name` environment variable).
+* `state`: The state of the column. One of `full`, `good`, `low`, `empty` and `offline`.
+* `dist_mm`: The measured distance from the top of the column to the first item.
+* `items`: The number of items in the column, determined from the measured distance and the column and item sizes given by environment variables.
+* `changes`: Indicates which fields changed.
+* `text`: A short description of what happened. In this case an item was dispensed.
+
+When the last item in a column is dispensed, the column becomes empty, as shown by the `state` and `items` values in the event:
+
+```json
+{ "col":1, "name": "cola", "state": "empty", "dist_mm": 811, "items": 0, "changes": "items,state", "text": "Item dispensed." }
+```
+
+When the vending machine is restocked, the column state returns to `full`:
+
+```json
+{ "col":1, "name":"cola", "state":"full", "dist_mm": 43, "items": 5, "changes": "items,state", "text": "Column restocked." }
+```
+
+When routine reporting is enabled by setting the `report_mins` environment variable, the state of each column is reported at that interval. A routine report doesn't have the `changes` property, and so can be easily distinguished from vending notifications.
+
+
+## Mock-up Vending Machine
+
+We didn't have a soda vending machine to hand, so we created a simple mockup that we used to test the app.
```

Comment:
```suggestion
We didn't have a soda vending machine on hand, so we created a simple mockup that we used to test the app.
```

You could consider rewording this whole sentence. The current wording makes it sound like we’re not serious enough to try this on a real machine, when we could instead make it seem like we’re trying to help the reader.

“If you’re looking to prototype this project and don’t have a real vending machine handy, here’s a simple mockup you can use to test the app.”